scripts/safeharbor/src/utils/generateSolidity.js

```diff
@@ -0,0 +1,56 @@
+// Generate solidity code from updates
+export function generateSolidityCode(updates) {
+    if (updates.length === 0) {
+        return "";
+    }
+
+    let solidityCode = `
+        // ---------- Bug Bounty Updates ----------
+        bytes[] memory calldatas = new bytes[](${updates.length});`;
+
+    updates.forEach((update, index) => {
+        solidityCode += `
+
+        // ${getDescription(update)}
+        calldatas[${index}] = "${update.calldata}";`;
+    });
+
+    solidityCode += `
+
+        _doSaferHarborUpdates(calldatas);
+    `;
+
+    return solidityCode.trim();
+}
+
+function getDescription(update) {
+    switch (update.function) {
+        case "removeChains": {
+            // Extracts and lists all chains to be removed.
+            const chains = update.args[0].join(", ");
+            return `Remove chains: ${chains}`;
+        }
+        case "addChains": {
+            // Describes adding contracts to new chains, listing addresses for each.
+            const descriptions = update.args[0].map(chainInfo => {
+                const accounts = chainInfo.accounts.map(acc => acc.accountAddress).join(", ");
```

Comment:
Not mandatory, but you could leverage [`Intl.ListFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat) instead of doing this manually.